newton/_src/utils/update_usd.py

```
@@ -0,0 +1,510 @@
+        self.time = round(time * self.fps)
+        self.stage.SetEndTimeCode(self.time)
+
+    def end_frame(self):
+        """
+        End the current frame.
+
+        This method is a placeholder for any end-of-frame logic required by the backend.
+        """
+        pass
+
+    def attach_source_stage(self, source_stage: str | Usd.Stage, output_stage: str | Usd.Stage):
```

Ruff (0.14.8)

Undefined name `Usd`

(F821)